core/src/test/java/org/htsjdk/core/utils/PathSpecifierUnitTest.java

```diff
@@ -0,0 +1,254 @@
+
+                {"path/to/localFile.bam",           "file://" + getCWD() + "path/to/localFile.bam", true, true},
+                {"/path/to/localFile.bam",          "file:///path/to/localFile.bam",    true, true},
+                {"file:path/to/localFile.bam",      "file:path/to/localFile.bam",       true, false},
+                {"file:/path/to/localFile.bam",     "file:/path/to/localFile.bam",      true, true},
+                {"file://path/to/localFile.bam",    "file://path/to/localFile.bam",     true, false}, // "path" looks like an authority, but won't be treated that way
+                {"file:///path/to/localFile.bam",   "file:///path/to/localFile.bam",    true, true},  // empty authority
+
+                {"gs://file.bam",                   "gs://file.bam",                    false, false},
+                {"gs://bucket/file.bam",            "gs://bucket/file.bam",             false, false},
+                {"gs:///bucket/file.bam",           "gs:///bucket/file.bam",            false, false},
+                {"gs://auth/bucket/file.bam",       "gs://auth/bucket/file.bam",        false, false},
+                {"gs://hellbender/test/resources/", "gs://hellbender/test/resources/",  false, false},
+
+                // uri must have a path: jimfs:file.bam
+                {"jimfs:file.bam",      "jimfs:file.bam", true, false},
+                // java.lang.AssertionError: java.net.URISyntaxException: Expected scheme-specific part at index 6: jimfs:
+                {"jimfs:/file.bam",     "jimfs:/file.bam", true, false},
+                // java.lang.AssertionError: uri must have a path: jimfs://file.bam
+                {"jimfs://file.bam",    "jimfs://file.bam", true, false},
+                // java.lang.AssertionError: java.net.URISyntaxException: Expected scheme-specific part at index 6: jimfs:
+                {"jimfs:///file.bam",   "jimfs:///file.bam", true, false},
+
+                //{"jimfs://root/file.bam",   "jimfs://root/file.bam", true, true},
+
+                // URIs with a "#" in the path part, which is valid URI syntax, but without encoding will be
+                // treated as a fragment delimiter.
+                {"/project/gvcf-pcr/23232_1#1/1.g.vcf.gz",      "file:///project/gvcf-pcr/23232_1%231/1.g.vcf.gz", true, true},
+                {"project/gvcf-pcr/23232_1#1/1.g.vcf.gz",       "file://" + getCWD() + "project/gvcf-pcr/23232_1%231/1.g.vcf.gz", true, true},
+
+                // URIs that are presented with a scheme included must already be escaped/encoded, otherwise we'd
+                // double-encode them
+                {"file:project/gvcf-pcr/23232_1#1/1.g.vcf.gz",  "file:project/gvcf-pcr/23232_1#1/1.g.vcf.gz", true, false},
+                {"file:/project/gvcf-pcr/23232_1#1/1.g.vcf.gz", "file:/project/gvcf-pcr/23232_1#1/1.g.vcf.gz", true, false},
+
+                {"gendb://somegdb",             "gendb://somegdb", false, false},
```

Comment:
I guess that this is the main reason to have the class in the first place (see the overall comment).

Reply:
See my overall comment describing the reasons for proposing these classes.